Add Hindi (hi) Translation

Anubhav-Mondal opened the pull request:
This pull request adds Hindi (hi_IN) language support to GoZen.

- Added hi.po file in the `src/translations`

- Registered Hindi in project.godot

- Updated localization.gd accordingly


Let me know if anything needs to be improved or if changes are required. I’m open to feedback 🙂

TheRealSlander commented:
First, I'd like to thank you for your contribution, it is greatly appreciated!

As you mentioned the translation to be for `hi_IN` (and not a general `hi` localization), can you rename the `hi.po` file to `hi_IN.po` to reflect what you said please?

If that's correct, can you modify the `project.godot` file accordingly as well to keep consistency.

Otherwise, I have some questions about some of your translations, can you check the following please:
```
 99 msgid "button_discord_server"
100 msgstr "डिस्कोर्ड सर्वर"
```
There we are talking about the Discord application/service, not the english "discord" word that means strike, friction or conflict. Can you ensure your translation is correct about it please? Maybe it is, I just want to be sure.

-----------------------------------------
```
170 msgid "text_gozen_made_by"
171 msgstr "GoZen एक वीडियो एडिटर है जिसे बनाया गया है"
```
There, the translation I have gives me:
`Gozen is a video editor made`
Where we would be expecting:
`Gozen is a video editor made by`
Can you check this is correct and not missing a word please?

-----------------------------------------
```
218 msgid "setting_tooltip_default_image_duration"
219 msgstr "फ़्रेम में अवधि।"
...
224 msgid "setting_tooltip_default_color_duration"
225 msgstr "फ़्रेम में अवधि।"
...
230 msgid "setting_tooltip_default_text_duration"
231 msgstr "फ़्रेम में अवधि।"
```
All these lines seem to have the same translation `Period in frame.`. Can you check this please?

-----------------------------------------
```
263 msgid "setting_tooltip_check_version"
264 msgstr "यह विकल्प GitHub रिपोजिटरी की जांच पर निर्भर करता है, और इसलिए इसके लिए इंटरनेट से कनेक्शन आवश्यक होता है। कुछ उपयोगकर्ता नहीं चाहते कि उनका ऐप इंटरनेट से जुड़े, इसलिए यह विकल्प डिफ़ॉल्ट रूप से 'अक्षम' होता है।"
```
The translation for this text gives me some weird things, talking about `Github Repiolent` (?). I'm pretty sure it should be `Github Repository` instead, maybe my translator is broken, but I just want to be sure this is not a typo from your side ^^.

-----------------------------------------
```
284 msgid "encoding_progress_text_sending_audio_error"
285 msgstr "ऑडियो एन्कोडर को भेजते समय कोई त्रुटि हुई!"
```
The translation to this seems to miss a word, as it translate to `There was an error while sending the audio encoder!` where it should be `There was an error while sending the audio to the encoder!`. The following line gives a correct translation, so maybe it is missing a word?
```
291 msgid "encoding_progress_text_sending_data_error"
292 msgstr "फ्रेम को एन्कोडर को भेजते समय कोई त्रुटि हुई!"
```

-----------------------------------------
I'll continue to check the translation after you confirmed/checked the above, as my post would be too long otherwise.

Again, thank you for your time and contribution.

Anubhav-Mondal commented:
Initially, I had Godot version `4.3` installed locally, and when I opened the project to register the `hi.po` file, it accidentally modified the `project.godot` version line to `4.3`. I didn’t notice at first, but I’ve now reverted it back to `4.4` to match the original setup.

Apologies for the hiccup — everything should be consistent now.

TheRealSlander commented:
Thanks for the precisions.

I  also forgot to mention the `hi` value inside the `hi_IN.po` (line 9) which should then be set to `hi_IN` as well to make things consistent.

Waiting for review/precisions about my previous posts to merge the localization to the main project. ;)

Anubhav-Mondal commented:
>218 msgid "setting_tooltip_default_image_duration"
>219 msgstr "फ़्रेम में अवधि।"
>...
>224 msgid "setting_tooltip_default_color_duration"
>225 msgstr "फ़्रेम में अवधि।"
>...
>230 msgid "setting_tooltip_default_text_duration"
>231 msgstr "फ़्रेम में अवधि।"
>
>All these lines seem to have the same translation Period in frame.. Can you check this please? 




I’ve verified that the original English source strings for those tooltip entries are all exactly the same:

`"setting_tooltip_default_image_duration" → The duration in frames.`
`"setting_tooltip_default_color_duration" → The duration in frames.`
`"setting_tooltip_default_text_duration" → The duration in frames.`

I’m also attaching a screenshot for reference in case it helps visualize the entries clearly.
![Screenshot 2025-07-07 232201](https://github.com/user-attachments/assets/be40ac30-85d5-4672-aa19-6726a73e90c4)


Since the English source uses the same phrase across all three, I translated them all as "फ़्रेम में अवधि।" to reflect that 1:1 in Hindi.

Let me know if anything needs to be changed.

TheRealSlander commented:
> In Hindi, the word "by" translates to "द्वारा", but unlike English, it's placed after the subject's name. So, for example:
> 
> “GoZen is a video editor made by Voylin” becomes “GoZen एक वीडियो एडिटर है जिसे बनाया गया है Voylin के द्वारा”
> 
> Adding “द्वारा” here without a name would make the sentence feel incomplete.

Ah! Ok, we already had this kind of problem with some other languages.

The adopted solution was to use a sentense like:
`GoZen is a video editor whose creator is`
or
`GoZen video editor creator:`
or
`Creator of the GoZen video editor:`

I let you find hwat suits the situation best. ^^'

Anubhav-Mondal commented:
>263 msgid "setting_tooltip_check_version"
>264 msgstr "यह विकल्प GitHub रिपोजिटरी की जांच पर निर्भर करता है, और इसलिए इसके लिए इंटरनेट से कनेक्शन आवश्यक होता है। कुछ उपयोगकर्ता नहीं चाहते कि उनका ऐप इंटरनेट से जुड़े, इसलिए यह विकल्प डिफ़ॉल्ट रूप से 'अक्षम' होता है।"
>
>The translation for this text gives me some weird things, talking about Github Repiolent (?). I'm pretty sure it should be Github >Repository instead, maybe my translator is broken, but I just want to be sure this is not a typo from your side ^^.

Yes, I think the translator might have glitched here. The word "रिपोजिटरी" is commonly used as a transliteration of "Repository" in Hindi tech contexts — it’s not a perfect translation, but widely understood in the dev community.

However, if you'd prefer, I can replace it with "Repository" directly (in English) for clarity and consistency.

Anubhav-Mondal commented:
>329 msgid "render_option_tooltip_gop_size"
>330 msgstr "Group Of Pictures। उच्च मान बेहतर कंप्रेशन देता है, लेकिन मान जितना अधिक होगा, वीडियो प्लेबैक के दौरान सटीकता (seeking और scrubbing) उतनी ही कम हो जाएगी। यदि आप इस वीडियो को आगे संपादन के लिए उपयोग करना चाहते हैं, तो मान को कम रखें।"
>It translates to:
>
>Group of Pictures. The higher value gives better compression, but the higher the value, the less will be as lowering accuracy during video playback. If you want to use this video for further editing, keep the value low.
>The ... the less will be as lowering accuracy during video playback ... part is what makes it weird... Can you ensure your translation >is correct please?


Yes, the original translation was mostly correct but a bit incomplete and slightly unclear.
I have improved the line, and now it reads more naturally and accurately in Hindi.

Anubhav-Mondal commented:
> Ah! Ok, we already had this kind of problem with some other languages.
> 
> The adopted solution was to use a sentense like: `GoZen is a video editor whose creator is` or `GoZen video editor creator:` or `Creator of the GoZen video editor:`
> 
> I let you find what best suits the situation. ^^'

I’ve fixed this issue. I translated the sentence `GoZen is a video editor whose creator is` to `GoZen एक वीडियो एडिटर है जिसका निर्माता है` at line 171. 
This structure now properly allows the subject name to be placed after the sentence, making it grammatically correct in Hindi.

Anubhav-Mondal commented:
The GOP message was already correct, but I’ve rephrased it to make it more straightforward and clear.

I’ve reviewed and addressed all the issues you pointed out.
If you notice anything else that needs fixing, I’m happy to take care of it.

voylin commented:
Thank you very much for your contribution! ^^
